pipenv/core.py

```diff
@@ -1561,20 +1561,18 @@ def pip_install(
     DEFAULT_EXISTS_ACTION = "w"
     if selective_upgrade:
         DEFAULT_EXISTS_ACTION = "i"
-    exists_action = vistir.misc.fs_str(
-        project.s.PIP_EXISTS_ACTION or DEFAULT_EXISTS_ACTION
-    )
+    exists_action = project.s.PIP_EXISTS_ACTION or DEFAULT_EXISTS_ACTION
```

Comment:
I think you might need to cast this one to a str otherwise it would be a boolean whereas it was a string before?

Reply:
I think we should explicitly look for a thruty string, since any string casted to bool will be True.

Reply:
It defaults to a string "w" https://github.com/pypa/pipenv/blob/main/pipenv/environments.py#L341

So if we leave it as a boolean where it is a string before then we should also review all usages of `PIP_EXISTS_ACTION`

Reply:
Neverminded, to my surprise this will apparently still be a string.
```
>>> "i" or "w"
'i'
```

Reply:
For now, leaving untouched. Will fix in #5063